src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/index.jsx

```diff
@@ -15,12 +16,38 @@ import styled from 'react-emotion';
 import space from 'app/styles/space';
 import {withTheme} from 'emotion-theming';
 
+const api = new Client();
+
 class SentryApplicationRow extends React.PureComponent {
   static propTypes = {
     app: PropTypes.object.isRequired,
     orgId: PropTypes.string.isRequired,
   };
 
+  redirectUrl = install => {
+    const {orgId, app} = this.props;
+
+    return app.redirectUrl
+      ? `${app.redirectUrl}?installationId=${install.uuid}&code=${install.code}`
+      : `/settings/${orgId}/integrations/`;
+  };
+
+  install = () => {
+    const {orgId, app} = this.props;
+
+    const success = install => browserHistory.push(this.redirectUrl(install));
+    const error = err => addErrorMessage(err.responseJSON);
+
+    const opts = {
+      method: 'POST',
+      data: {slug: app.slug},
+      success,
+      error,
+    };
+
+    api.request(`/organizations/${orgId}/sentry-app-installations/`, opts);
```

Comment:
You could use `api.requestPromise()`. It would let you use `then(success, error)` to save two lines 🏌️

Reply:
I had that originally, but couldn't figure out how to make it work in tests, tbth. haha. The assertions would run before the promise would resolves. 😶

Reply:
You have to use `await wrapper.update()`